core/common/src/MapLayerSettings.ts

```diff
@@ -198,6 +199,16 @@ export interface MapLayerKey {
   value: string;
 }
 
+/**
+   * stores key-value pair to be added to all requests made involving map layer.
+   * @beta
+   */
+export interface MapLayerUrlParam {
+  key: string;
+  value: string;
+  secret?: boolean;
```

Comment:
@pmconne @calebmshafer  MapLayerUrlParam represent one `key=value` pair that should be apprend to map layer base URL.  This can contain non-secret information such as: `apiVersion=2`.  But it might contains some nasty information such as `password=12345`.

What do you think of having a `secret `property, when turned on, the param would not get serialized part of the JSON representation.  That we leave the decision on how to handle the secret to the API caller.
Do you foresee potential issues?  Should we have a more brute approach and strip all custom parameters from serialized json?

Reply:
Is there any reason for us to know/care about individual query parameters?
Doesn't the problem boil down to: we need a URL which may contain query parameters, and some of those query parameters should not be persisted?
If so, can't you just accept a URL to persist, plus optional additional query params to append but not persist?
If not, can you clarify (in your PR description or an issue linked therefrom) what problem you are trying to solve?

Reply:
The "url' parameter we see everywhere is in fact the **service endpoint** URL which means any query parameters get discarded no matter what, so if you pass:
@pmconne 
https://server/service?customParam=value => https://server/service?<Service_official_parameters_appended_here>

I think this behavior is ok (although it is somewhat unexpected given the actual documentation).
That's why I suggest adding a `customParameters` property where the user would be free to pass any custom parameters (which of course should not conflict with the service standard parameters).  The idea of having a `isSecret` prop, is for case where user want to use custom parameter like username=password=123.

Reply:
Yeah that's confusing that we automatically strip out all query parameters. We can't change the behavior now but we should fix the documentation.

`isSecret` isn't an accurate name for a property that really means "don't store this parameter". And because we currently don't save any query parameters, that should be default for the new parameters too.

Did you consider the following instead of a single `customParameters` array?
```
{
  savedQueryParams?: { [key: string]: string },
  unsavedQueryParams?: { [key: string]: string },
}
```

Reply:
That could work yes.. what is the benefit of using an Index signature here?  Do you want to express the uniqueness of param keys ?  If so, that could still happen be because we have two independents list.

Reply:
> benefit of using an Index signature

Because these are key-value pairs of strings, rather than inventing a new type `MapLayerUrlParam` for such a simple concept.

Reply:
Please publish it so we can see CI results.